picard/formats/__init__.py

```diff
@@ -45,7 +45,7 @@ def supported_extensions():
 def guess_format(filename, options=_formats):
     """Select the best matching file type amongst supported formats."""
     results = []
-    with file(filename, "rb") as fileobj:
+    with file(filename, "rb", 128) as fileobj:
```

Comment:
I fear it would have a counter effect. Actually it may be much slower. Therefore please profile this code (that is using buffer of 128 instead of specifying none which is equal to using system default).

See https://docs.python.org/2/library/functions.html#open and http://stackoverflow.com/a/29712601